pkg/apis/pipeline/v1beta1/taskrun_validation.go

```diff
@@ -93,6 +94,22 @@ func (ts *TaskRunSpec) Validate(ctx context.Context) (errs *apis.FieldError) {
 	return errs
 }
 
+// validateStepComputeResources validates compute resources are not configured in TaskRunSpec.Step
+func validateStepComputeResources(computeResources *corev1.ResourceRequirements, taskSpec *TaskSpec) (errs *apis.FieldError) {
+	if taskSpec == nil {
+		return nil
+	}
```

Comment:
I'd prefer for this logic to be in the reconciler rather than the webhook. The reason is that the reconciler resolves the Task spec that will actually be used for the TaskRun, whether that comes from TaskRun.TaskSpec or TaskRun.TaskRef. This validation needs to happen for TaskRuns that use a TaskRef as well.